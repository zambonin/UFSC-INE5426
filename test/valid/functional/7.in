int ref t[10]
int ref output
output = fold(lambda int ref x, y -> x + y, t)
