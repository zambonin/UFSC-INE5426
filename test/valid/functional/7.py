exec(open('../../../src/scope_manager.py', 'r').read())
t = [0] * 10

def t_fold(t):
  def λ(x, y):
    return (x + y)

    
  t_tv = t[0]
  
  s_context()
  t_ti = 1
  while (t_ti < len(t)):
    t_tv = (t_tv + λ(t_tv, t[t_ti]))
    t_ti = (t_ti + 1)
  r_context()
  return t_tv

output = t_fold(t)
