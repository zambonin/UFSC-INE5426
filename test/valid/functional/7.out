int ref array: t (size: 10)
int ref var: output
int ref fun: t_fold (params: int ref array t)
  int ref fun: lambda (params: int ref x, int ref y)
    ret + x y
  int ref var: t_tv
  = t_tv [index] t 0
  int var: t_ti
  for: = t_ti 1, < t_ti [len] t, = t_ti + t_ti 1
  do:
    = t_tv + t_tv lambda[2 params] t_tv [index] t t_ti
  ret t_tv
= output t_fold[1 params] t
